Issue upgrading 1.45.2 -> 1.46.0

I have the following docker-compose section:
```
  mockgcs:
    image: fsouza/fake-gcs-server:1.45.2
    ports:
      - "4443:4443"
    volumes:
      - ./mockgcs/buckets:/storage
    command: -scheme http -public-host localhost
    deploy:
      resources:
        limits:
          memory: 128M
```

I have a folder located in `./mockgcs/buckets/` called `bucket-name`.

I can fetch the bucket objects by visiting: `http://localhost:4443/storage/v1/b/bucket-name/o`

When I change the version 1.46.0. I can no longer view this bucket url or post to it, I am given the following response:
```
{"error":{"code":404,"message":"Not Found","errors":null}}
```

I have been getting similar issue when I use 
```
 mockgcs:
    image: fsouza/fake-gcs-server
    ports:
      - "4443:4443"
```
The error received in the spring application which tries to upload the file in the bucket is
```
com.google.cloud.storage.StorageException
 Caused by: java.io.IOException
```
However, when I use the version 1.45.2, it works fine

--- a/internal/backend/fs.go
+++ b/internal/backend/fs.go
@@ -159,6 +159,9 @@ func (s *storageFS) GetBucket(name string) (Bucket, error) {
 func getBucketAttributes(path string) (BucketAttrs, error) {
 	content, err := os.ReadFile(path + bucketMetadataSuffix)
 	if err != nil {
+		if os.IsNotExist(err) {
+			return BucketAttrs{}, nil
+		}
 		return BucketAttrs{}, err
 	}
 	var attrs BucketAttrs

--- a/internal/backend/fs_test.go
+++ b/internal/backend/fs_test.go
@@ -1,0 +1,76 @@
+// Copyright 2023 Francisco Souza. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package backend
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestGetAttributes(t *testing.T) {
+	t.Parallel()
+
+	tempDir := t.TempDir()
+	testBucket := filepath.Join(tempDir, "some-bucket")
+	bucketAttrs := BucketAttrs{
+		DefaultEventBasedHold: false,
+		VersioningEnabled:     true,
+	}
+	data, _ := json.Marshal(bucketAttrs)
+	err := os.WriteFile(testBucket+bucketMetadataSuffix, data, 0o644)
+	if err != nil {
+		t.Fatal(err)
+	}
+	notABucket := filepath.Join(tempDir, "not-a-bucket")
+	err = os.WriteFile(notABucket+bucketMetadataSuffix, []byte("this is not valid json"), 0o644)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name          string
+		inputPath     string
+		expectedAttrs BucketAttrs
+		expectErr     bool
+	}{
+		{
+			name:      "file not found",
+			inputPath: filepath.Join(tempDir, "unknown-bucket"),
+		},
+		{
+			name:          "existing bucket",
+			inputPath:     testBucket,
+			expectedAttrs: bucketAttrs,
+		},
+		{
+			name:      "invalid file",
+			inputPath: notABucket,
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			attrs, err := getBucketAttributes(test.inputPath)
+			if test.expectErr && err == nil {
+				t.Fatal("expected error, but got <nil>")
+			}
+
+			if !test.expectErr && err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			if diff := cmp.Diff(attrs, test.expectedAttrs); diff != "" {
+				t.Errorf("incorrect attributes returned\nwant: %#v\ngot:  %#v\ndiff: %s", test.expectedAttrs, attrs, diff)
+			}
+		})
+	}
+}

